### PR TITLE
[Storage] Conditional etag parameters

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py [hunk 1]
```diff
@@ -47,3 +50,47 @@ def get_page_ranges_result(ranges):
...
+    # type: (Dict[str, Any]) -> ModifiedAccessConditions
+    if_match = None
+    if_none_match = None
+    match_condition = kwargs.pop('match_condition', None)
+    if match_condition == MatchConditions.IfNotModified:
+        if_match = kwargs.pop('etag', None)
+    elif match_condition == MatchConditions.IfPresent:
+        if_match = '*'
+    elif match_condition == MatchConditions.IfModified:
+        if_none_match = kwargs.pop('etag', None)
```
Comment (johanste) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: This should raise if no `etag` is provided

Comment (johanste) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: Same with ifNotModified.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py [hunk 2]
```diff
@@ -47,3 +50,47 @@ def get_page_ranges_result(ranges):
...
+
+    return ModifiedAccessConditions(
+        if_modified_since=kwargs.pop('if_modified_since', None),
+        if_unmodified_since=kwargs.pop('if_unmodified_since', None),
+        if_match=if_match or kwargs.pop('if_match', None),
+        if_none_match=if_none_match or kwargs.pop('if_none_match', None)
+    )
+
+
+def get_source_conditions(kwargs):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: Could we please create a _serialize.py and put these methods there?

Comment (annatisch) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: Done :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py (hunk 1 above)
Comment (annatisch) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: Fixed :)

**Pull request merged**